src/scripts/main.js

```diff
@@ -1,3 +1,253 @@
+    e.target.textContent = 'Restart';
+    messageStart.classList.toggle('hidden');
+    maxNum = 0;
+    noEmpty = false;
+    score.innerText = maxNum;
+    randomNum();
+    paint();
+  } else {
+    e.target.textContent = 'Start';
+    messageStart.classList.toggle('hidden');
+
+    arrOfNumbers = [
+      ['', '', '', ''],
+      ['', '', '', ''],
+      ['', '', '', ''],
+      ['', '', '', ''],
+    ];
+    maxNum = 0;
+    noEmpty = false;
+    score.innerText = maxNum;
+    moves = 0;
+    paint();
+  }
+});
+
+function hasEmpty() {
+  for (const item of arrOfNumbers) {
+    if (Math.max(...item) >= 2048) {
+      messageWin.classList.toggle('hidden');
+    }
+  }
+
+  for (let i = 0; i < 4; i++) {
+    for (let j = 0; j < 4; j++) {
+      if (arrOfNumbers[i][j] === '') {
+        return true;
+      }
+    }
+  }
+  noEmpty = true;
+
+  return false;
+}
+
+function randomWithProbability() {
+  const notRandomNumbers = [2, 2, 2, 2, 2, 2, 2, 2, 2, 4];
+  const idx = Math.floor(Math.random() * notRandomNumbers.length);
+
+  return notRandomNumbers[idx];
+}
+
+function randomNum() {
+  if (!hasEmpty()) {
+    return;
+  }
+
+  let x;
+  let y;
+  const value = randomWithProbability();
+
+  if (moves === 0) {
+    x = Math.floor(Math.random() * 4);
+    y = Math.floor(Math.random() * 4);
+
+    const x1 = Math.floor(Math.random() * 4);
+    const y1 = Math.floor(Math.random() * 4);
+
+    arrOfNumbers[x][y] = value;
+    arrOfNumbers[x1][y1] = value;
+    moves++;
+  } else {
+    x = Math.floor(Math.random() * 4);
+    y = Math.floor(Math.random() * 4);
+
+    while (arrOfNumbers[x][y] !== '') {
+      x = Math.floor(Math.random() * 4);
+      y = Math.floor(Math.random() * 4);
+    }
+  }
+
+  arrOfNumbers[x][y] = value;
+  moves += 1;
+};
+
+function paint() {
+  if (noEmpty) {
+    mes.innerHTML = `<p class="message message-lose">
+    You lose! Restart the game?</p>`;
+
+    return;
+  }
+
+  for (let i = 0; i < arrOfNumbers.length; i++) {
+    for (let k = 0; k < arrOfNumbers[i].length; k++) {
+      const item = arrOfNumbers[i][k];
+
+      if (item !== '') {
+        rows[i].cells[k].innerText = item;
+        rows[i].cells[k].classList = `field-cell field-cell--${item}`;
+      }
+
+      if (item === '') {
+        rows[i].cells[k].innerText = '';
+        rows[i].cells[k].classList = `field-cell`;
+      }
+    }
+  }
+};
+
+function filterZero(row) {
+  const clearRow = row.filter(num => num !== '');
+
+  return clearRow;
+};
+
+function slide(row) {
+  let clearRow = filterZero(row);
+
+  for (let i = 0; i < clearRow.length - 1; i++) {
+    if (clearRow[i] === clearRow[i + 1]) {
+      clearRow[i] *= 2;
+      maxNum += clearRow[i];
+      clearRow[i + 1] = '';
+      gameScore.innerText = maxNum;
+    }
+  }
+
+  clearRow = filterZero(clearRow);
+
+  while (clearRow.length < 4) {
+    clearRow.push('');
+  }
+
+  return clearRow;
+}
+
+document.addEventListener('keyup', (e) => {
+  if (e.code === 'ArrowLeft') {
```

Comment:
Move all directions to the object.

Example of usage: 
`if (e.code === Directions.ArrowLeft)`